src/extension-runners/chromium.js

```diff
@@ -0,0 +1,266 @@
+
+  async createReloadManagerExtension() {
+    const tmpDir = new TempDir();
+    await tmpDir.create();
+    this.registerCleanup(() => tmpDir.remove());
+
+    const extPath = path.join(
+      tmpDir.path(),
+      `reload-manager-extension-${Date.now()}`
+    );
+
+    log.debug(`Creating reload-manager-extension in ${extPath}`);
+
+    await asyncMkdirp(extPath);
+
+    await fs.writeFile(
+      path.join(extPath, 'manifest.json'),
+      JSON.stringify({
+        manifest_version: 2,
+        name: 'web-ext Reload Manager Extension',
+        version: '1.0',
+        permissions: ['management', 'tabs'],
+        background: {
+          scripts: ['bg.js'],
+        },
+      })
+    );
+
+    const bgPage = `function bgPage(webExtParams) {
+      async function getAllDevAddons() {
+        const [managerAddon, allAddons] = await Promise.all([
+          new Promise(r => chrome.management.getSelf(r)),
+          new Promise(r => chrome.management.getAll(r)),
+        ]);
+
+        return allAddons.filter((addon) => {
+          return addon.installType === "development" &&
+            addon.id !== managerAddon.id;
+        });
+      }
+
+      const setEnabled = (addonId, value) =>
+        new Promise(r => chrome.management.setEnabled(addonId, value, r));
+
+      async function reloadAddon(addonId) {
+        await setEnabled(addonId, false);
+        await setEnabled(addonId, true);
+      }
+
+      const ws = new window.WebSocket([
+        "ws://", webExtParams.address, ":", webExtParams.port
+      ].join(""));
+
+      ws.onmessage = async (evt) => {
+        const msg = JSON.parse(evt.data);
+        if (msg.type === 'webExtReloadAllExtensions') {
+          const {extension} = msg;
+
+          const devExtensions = await getAllDevAddons();
+          for (var ext of devExtensions) {
+            reloadAddon(ext.id);
+          }
+        }
+      };
+    }`;
+
+    await fs.writeFile(path.join(extPath, 'bg.js'),
+                       `(${bgPage})(${JSON.stringify(this.wss.address())})`
+    );
+    return extPath;
+  }
+
+  /**
+   * Reloads all the extensions, collect any reload error and resolves to
+   * an array composed by a single ExtensionRunnerReloadResult object.
+   */
+  async reloadAllExtensions(): Promise<Array<ExtensionRunnerReloadResult>> {
+    const runnerName = this.getName();
+
+    this.wssBroadcast({
+      type: 'webExtReloadAllExtensions',
+    });
+
+    return [{runnerName}];
+  }
+
+  /**
+   * Reloads a single extension, collect any reload error and resolves to
+   * an array composed by a single ExtensionRunnerReloadResult object.
+   */
+  async reloadExtensionBySourceDir(
+    extensionSourceDir: string // eslint-disable-line no-unused-vars
+  ): Promise<Array<ExtensionRunnerReloadResult>> {
+    // TODO(rpl): file github issue to improve the manager extension
+    // to make it able to detect the extension ids assigned to the
+    // target extensions and map it to the extensions source dir.
```

Comment:
The extension ID of a Chrome extension can be determined without helper extension:
It may be set through `manifest.json` (via the `key` attribute). If unset, one is automatically generated based on the file path (see https://cs.chromium.org/Extension::InitExtensionID%20file:cc ).

Reply:
Noted, we may consider that option once we look into this improvement in a follow up.

Reply:
Let's add a comment to my comment above in the code, so we don't forget about it.

Reply:
also filed as #1687